prov/rxm: build warnings on windows

Snippet from AppVeyor logs:

prov\rxd\src\rxd_init.c(38): warning C4459: declaration of 'rxd_info' hides global declaration [C:\projects\libfabric\libfabric.vcxproj]
  c:\projects\libfabric\prov\rxd\src\rxd.h(94): note: see declaration of 'rxd_info'
  rxd_rma.c
C:\projects\libfabric\include\windows/osd.h(863): warning C4244: '=': conversion from 'int' to 'float', possible loss of data [C:\projects\libfabric\libfabric.vcxproj]
  rxm.c
C:\projects\libfabric\include\windows/osd.h(863): warning C4244: '=': conversion from 'int' to 'float', possible loss of data [C:\projects\libfabric\libfabric.vcxproj]
  rxm_attr.c
C:\projects\libfabric\include\windows/osd.h(863): warning C4244: '=': conversion from 'int' to 'float', possible loss of data [C:\projects\libfabric\libfabric.vcxproj]
prov\rxm\src\rxm_attr.c(44): warning C4245: 'initializing': conversion from 'int' to 'uint64_t', signed/unsigned mismatch [C:\projects\libfabric\libfabric.vcxproj]
prov\rxm\src\rxm_attr.c(52): warning C4245: 'initializing': conversion from 'int' to 'uint64_t', signed/unsigned mismatch [C:\projects\libfabric\libfabric.vcxproj]
  rxm_conn.c
C:\projects\libfabric\include\windows/osd.h(863): warning C4244: '=': conversion from 'int' to 'float', possible loss of data [C:\projects\libfabric\libfabric.vcxproj]
prov\rxm\src\rxm_conn.c(230): warning C4389: '!=': signed/unsigned mismatch [C:\projects\libfabric\libfabric.vcxproj]
  rxm_rma.c
C:\projects\libfabric\include\windows/osd.h(863): warning C4244: '=': conversion from 'int' to 'float', possible loss of data [C:\projects\libfabric\libfabric.vcxproj]
prov\rxm\src\rxm_rma.c(75): warning C4244: 'return': conversion from 'ssize_t' to 'int', possible loss of data [C:\projects\libfabric\libfabric.vcxproj]
prov\rxm\src\rxm_rma.c(173): warning C4244: 'return': conversion from 'ssize_t' to 'int', possible loss of data [C:\projects\libfabric\libfabric.vcxproj]
prov\rxm\src\rxm_rma.c(178): warning C4244: 'return': conversion from 'ssize_t' to 'int', possible loss of data [C:\projects\libfabric\libfabric.vcxproj]
prov\rxm\src\rxm_rma.c(218): warning C4244: '=': conversion from 'ssize_t' to 'int', possible loss of data [C:\projects\libfabric\libfabric.vcxproj]
  rxm_cq.c
C:\projects\libfabric\include\windows/osd.h(863): warning C4244: '=': conversion from 'int' to 'float', possible loss of data [C:\projects\libfabric\libfabric.vcxproj]
prov\rxm\src\rxm_cq.c(180): warning C4244: '=': conversion from 'int' to 'uint8_t', possible loss of data [C:\projects\libfabric\libfabric.vcxproj]
prov\rxm\src\rxm_cq.c(222): warning C4244: '=': conversion from 'ssize_t' to 'int', possible loss of data [C:\projects\libfabric\libfabric.vcxproj]
prov\rxm\src\rxm_cq.c(256): warning C4244: '=': conversion from 'uint64_t' to 'int', possible loss of data [C:\projects\libfabric\libfabric.vcxproj]
prov\rxm\src\rxm_cq.c(437): warning C4244: '=': conversion from 'ssize_t' to 'int', possible loss of data [C:\projects\libfabric\libfabric.vcxproj]
prov\rxm\src\rxm_cq.c(598): warning C4018: '<': signed/unsigned mismatch [C:\projects\libfabric\libfabric.vcxproj]
  Compiling...
  rxm_domain.c
C:\projects\libfabric\include\windows/osd.h(863): warning C4244: '=': conversion from 'int' to 'float', possible loss of data [C:\projects\libfabric\libfabric.vcxproj]
  rxm_ep.c
C:\projects\libfabric\include\windows/osd.h(863): warning C4244: '=': conversion from 'int' to 'float', possible loss of data [C:\projects\libfabric\libfabric.vcxproj]
prov\rxm\src\rxm_ep.c(164): warning C4244: '=': conversion from 'int' to 'uint8_t', possible loss of data [C:\projects\libfabric\libfabric.vcxproj]
prov\rxm\src\rxm_ep.c(175): warning C4244: 'function': conversion from 'int' to 'uint8_t', possible loss of data [C:\projects\libfabric\libfabric.vcxproj]
prov\rxm\src\rxm_ep.c(289): warning C4244: '=': conversion from 'ssize_t' to 'int', possible loss of data [C:\projects\libfabric\libfabric.vcxproj]
prov\rxm\src\rxm_ep.c(521): warning C4267: '=': conversion from 'size_t' to 'uint8_t', possible loss of data [C:\projects\libfabric\libfabric.vcxproj]
prov\rxm\src\rxm_ep.c(666): warning C4267: '=': conversion from 'size_t' to 'uint8_t', possible loss of data [C:\projects\libfabric\libfabric.vcxproj]
prov\rxm\src\rxm_ep.c(706): warning C4267: '=': conversion from 'size_t' to 'uint8_t', possible loss of data [C:\projects\libfabric\libfabric.vcxproj]
prov\rxm\src\rxm_ep.c(717): warning C4244: '=': conversion from 'int' to 'uint8_t', possible loss of data [C:\projects\libfabric\libfabric.vcxproj]
prov\rxm\src\rxm_ep.c(752): warning C4244: '=': conversion from 'ssize_t' to 'int', possible loss of data [C:\projects\libfabric\libfabric.vcxproj]
prov\rxm\src\rxm_ep.c(773): warning C4244: '=': conversion from 'ssize_t' to 'int', possible loss of data [C:\projects\libfabric\libfabric.vcxproj]
prov\rxm\src\rxm_ep.c(787): warning C4244: '=': conversion from 'ssize_t' to 'int', possible loss of data [C:\projects\libfabric\libfabric.vcxproj]
  rxm_fabric.c
C:\projects\libfabric\include\windows/osd.h(863): warning C4244: '=': conversion from 'int' to 'float', possible loss of data [C:\projects\libfabric\libfabric.vcxproj]

--- a/include/fi.h
+++ b/include/fi.h
@@ -163,7 +163,7 @@ static inline size_t fi_get_aligned_sz(size_t size, size_t alignment)
 
 uint64_t fi_tag_bits(uint64_t mem_tag_format);
 uint64_t fi_tag_format(uint64_t tag_bits);
-int fi_size_bits(uint64_t num);
+uint8_t fi_size_bits(uint64_t num);
 
 int ofi_send_allowed(uint64_t caps);
 int ofi_recv_allowed(uint64_t caps);

--- a/include/fi_mem.h
+++ b/include/fi_mem.h
@@ -56,9 +56,9 @@
 /* We implement memdup to avoid external library dependency */
 static inline void *mem_dup(const void *src, size_t size)
 {
-	void *dest;
+	void *dest = malloc(size);
 
-	if ((dest = malloc(size)))
+	if (dest)
 		memcpy(dest, src, size);
 	return dest;
 }

--- a/include/rdma/fabric.h
+++ b/include/rdma/fabric.h
@@ -229,18 +229,18 @@ enum fi_resource_mgmt {
 	FI_RM_ENABLED
 };
 
-#define FI_ORDER_NONE		0
-#define FI_ORDER_RAR		(1 << 0)
-#define FI_ORDER_RAW		(1 << 1)
-#define FI_ORDER_RAS		(1 << 2)
-#define FI_ORDER_WAR		(1 << 3)
-#define FI_ORDER_WAW		(1 << 4)
-#define FI_ORDER_WAS		(1 << 5)
-#define FI_ORDER_SAR		(1 << 6)
-#define FI_ORDER_SAW		(1 << 7)
-#define FI_ORDER_SAS		(1 << 8)
+#define FI_ORDER_NONE		0ULL
+#define FI_ORDER_RAR		(1ULL << 0)
+#define FI_ORDER_RAW		(1ULL << 1)
+#define FI_ORDER_RAS		(1ULL << 2)
+#define FI_ORDER_WAR		(1ULL << 3)
+#define FI_ORDER_WAW		(1ULL << 4)
+#define FI_ORDER_WAS		(1ULL << 5)
+#define FI_ORDER_SAR		(1ULL << 6)
+#define FI_ORDER_SAW		(1ULL << 7)
+#define FI_ORDER_SAS		(1ULL << 8)
 #define FI_ORDER_STRICT		0x1FF
-#define FI_ORDER_DATA		(1 << 16)
+#define FI_ORDER_DATA		(1ULL << 16)
 
 enum fi_ep_type {
 	FI_EP_UNSPEC,

--- a/prov/rxm/src/rxm.h
+++ b/prov/rxm/src/rxm.h
@@ -290,7 +290,7 @@ struct rxm_recv_queue {
 struct rxm_buf_pool {
 	struct util_buf_pool *pool;
 	struct dlist_entry buf_list;
-	uint8_t local_mr;
+	int local_mr;
 	fastlock_t lock;
 };
 
@@ -355,7 +355,7 @@ int rxm_domain_open(struct fid_fabric *fabric, struct fi_info *info,
 int rxm_cq_open(struct fid_domain *domain, struct fi_cq_attr *attr,
 			 struct fid_cq **cq_fid, void *context);
 void rxm_cq_progress(struct rxm_ep *rxm_ep);
-int rxm_cq_handle_data(struct rxm_rx_buf *rx_buf);
+ssize_t rxm_cq_handle_data(struct rxm_rx_buf *rx_buf);
 
 int rxm_endpoint(struct fid_domain *domain, struct fi_info *info,
 			  struct fid_ep **ep, void *context);

--- a/prov/rxm/src/rxm_attr.c
+++ b/prov/rxm/src/rxm_attr.c
@@ -43,15 +43,15 @@
 
 struct fi_tx_attr rxm_tx_attr = {
 	.caps = RXM_EP_CAPS,
-	.msg_order = ~0x0,
-	.comp_order = ~0x0,
+	.msg_order = ~0x0ULL,
+	.comp_order = ~0x0ULL,
 	.size = SIZE_MAX,
 	.iov_limit = RXM_IOV_LIMIT,
 };
 
 struct fi_rx_attr rxm_rx_attr = {
 	.caps = RXM_EP_CAPS,
-	.msg_order = ~0x0,
+	.msg_order = ~0x0ULL,
 	.comp_order = FI_ORDER_STRICT | FI_ORDER_DATA,
 	.size = 1024,
 	.iov_limit= RXM_IOV_LIMIT,

--- a/prov/rxm/src/rxm_conn.c
+++ b/prov/rxm/src/rxm_conn.c
@@ -227,7 +227,7 @@ static void *rxm_conn_event_handler(void *arg)
 			break;
 		case FI_CONNREQ:
 			FI_DBG(&rxm_prov, FI_LOG_FABRIC, "Got new connection\n");
-			if (rd != len) {
+			if ((size_t)rd != len) {
 				FI_WARN(&rxm_prov, FI_LOG_FABRIC,
 					"Received size (%zd) not matching "
 					"expected (%zu)\n", rd, len);
@@ -308,7 +308,9 @@ rxm_conn_connect(struct util_ep *util_ep, struct util_cmap_handle *handle,
 
 	free(rxm_ep->msg_info->dest_addr);
 	rxm_ep->msg_info->dest_addrlen = addrlen;
-	if (!(rxm_ep->msg_info->dest_addr = mem_dup(addr, rxm_ep->msg_info->dest_addrlen)))
+
+	rxm_ep->msg_info->dest_addr = mem_dup(addr, rxm_ep->msg_info->dest_addrlen);
+	if (!rxm_ep->msg_info->dest_addr)
 		return -FI_ENOMEM;
 
 	ret = fi_getinfo(rxm_ep->util_ep.domain->fabric->fabric_fid.api_version,

--- a/prov/rxm/src/rxm_cq.c
+++ b/prov/rxm/src/rxm_cq.c
@@ -155,7 +155,7 @@ static int rxm_match_iov(const struct iovec *iov, void **desc,
 			 uint8_t count, uint64_t offset, size_t match_len,
 			 struct rxm_iov *match_iov)
 {
-	int i;
+	uint8_t i;
 
 	assert(count <= RXM_IOV_LIMIT);
 
@@ -178,7 +178,7 @@ static int rxm_match_iov(const struct iovec *iov, void **desc,
 	}
 	assert(!match_len);
 	match_iov->count = i + 1;
-	return 0;
+	return FI_SUCCESS;
 }
 
 static int rxm_match_rma_iov(struct rxm_recv_entry *recv_entry,
@@ -212,10 +212,10 @@ static int rxm_match_rma_iov(struct rxm_recv_entry *recv_entry,
 	return 0;
 }
 
-static int rxm_lmt_rma_read(struct rxm_rx_buf *rx_buf)
+static ssize_t rxm_lmt_rma_read(struct rxm_rx_buf *rx_buf)
 {
 	struct rxm_iov *match_iov = &rx_buf->match_iov[rx_buf->index];
-	int ret;
+	ssize_t ret;
 
 	ret = fi_readv(rx_buf->conn->msg_ep, match_iov->iov, match_iov->desc,
 		       match_iov->count, 0, rx_buf->rma_iov->iov[0].addr,
@@ -246,7 +246,7 @@ static int rxm_lmt_tx_finish(struct rxm_tx_entry *tx_entry)
 static int rxm_lmt_handle_ack(struct rxm_rx_buf *rx_buf)
 {
 	struct rxm_tx_entry *tx_entry;
-	int index;
+	uint64_t index;
 
 	FI_DBG(&rxm_prov, FI_LOG_CQ, "Got ACK for msg_id: 0x%" PRIx64 "\n",
 			rx_buf->pkt.ctrl_hdr.msg_id);
@@ -271,7 +271,7 @@ static int rxm_lmt_handle_ack(struct rxm_rx_buf *rx_buf)
 	}
 }
 
-int rxm_cq_handle_data(struct rxm_rx_buf *rx_buf)
+ssize_t rxm_cq_handle_data(struct rxm_rx_buf *rx_buf)
 {
 	struct rxm_iov mr_match_iov;
 	size_t i, rma_total_len = 0;
@@ -339,7 +339,7 @@ int rxm_cq_handle_data(struct rxm_rx_buf *rx_buf)
 	}
 }
 
-int rxm_handle_recv_comp(struct rxm_rx_buf *rx_buf)
+ssize_t rxm_handle_recv_comp(struct rxm_rx_buf *rx_buf)
 {
 	struct rxm_recv_match_attr match_attr;
 	struct dlist_entry *entry;
@@ -400,11 +400,11 @@ int rxm_handle_recv_comp(struct rxm_rx_buf *rx_buf)
 	return rxm_cq_handle_data(rx_buf);
 }
 
-static int rxm_lmt_send_ack(struct rxm_rx_buf *rx_buf)
+static ssize_t rxm_lmt_send_ack(struct rxm_rx_buf *rx_buf)
 {
 	struct rxm_tx_entry *tx_entry;
 	struct rxm_tx_buf *tx_buf;
-	int ret;
+	ssize_t ret;
 
 	assert(rx_buf->conn);
 
@@ -414,7 +414,8 @@ static int rxm_lmt_send_ack(struct rxm_rx_buf *rx_buf)
 		return -FI_EAGAIN;
 	}
 
-	if (!(tx_entry = rxm_tx_entry_get(&rx_buf->ep->send_queue))) {
+	tx_entry = rxm_tx_entry_get(&rx_buf->ep->send_queue);
+	if (!tx_entry) {
 		ret = -FI_EAGAIN;
 		goto err1;
 	}
@@ -466,8 +467,8 @@ static int rxm_handle_remote_write(struct rxm_ep *rxm_ep,
 	return 0;
 }
 
-static int rxm_cq_handle_comp(struct rxm_ep *rxm_ep,
-			      struct fi_cq_tagged_entry *comp)
+static ssize_t rxm_cq_handle_comp(struct rxm_ep *rxm_ep,
+				  struct fi_cq_tagged_entry *comp)
 {
 	enum rxm_proto_state state = RXM_GET_PROTO_STATE(comp);
 	struct rxm_rx_buf *rx_buf = comp->op_context;
@@ -579,7 +580,8 @@ static ssize_t rxm_cq_read(struct fid_cq *msg_cq, struct fi_cq_tagged_entry *com
 void rxm_cq_progress(struct rxm_ep *rxm_ep)
 {
 	struct fi_cq_tagged_entry comp;
-	ssize_t ret, comp_read = 0;
+	ssize_t ret;
+	size_t comp_read = 0;
 
 	do {
 		ret = rxm_cq_read(rxm_ep->msg_cq, &comp);
@@ -590,7 +592,7 @@ void rxm_cq_progress(struct rxm_ep *rxm_ep)
 			goto err;
 
 		if (ret) {
-			comp_read += ret;
+			comp_read += (size_t)ret;
 			ret = rxm_cq_handle_comp(rxm_ep, &comp);
 			if (ret)
 				goto err;

--- a/prov/rxm/src/rxm_domain.c
+++ b/prov/rxm/src/rxm_domain.c
@@ -109,7 +109,8 @@ static int rxm_mr_reg(struct fid *domain_fid, const void *buf, size_t len,
 	rxm_domain = container_of(domain_fid, struct rxm_domain,
 			util_domain.domain_fid.fid);
 
-	if (!(rxm_mr = calloc(1, sizeof(*rxm_mr))))
+	rxm_mr = calloc(1, sizeof(*rxm_mr));
+	if (!rxm_mr)
 		return -FI_ENOMEM;
 
 	/* Additional flags to use RMA read for large message transfers */

--- a/prov/rxm/src/rxm_ep.c
+++ b/prov/rxm/src/rxm_ep.c
@@ -278,18 +278,18 @@ int rxm_ep_repost_buf(struct rxm_rx_buf *rx_buf)
 {
 	struct rxm_buf hdr = rx_buf->hdr;
 	struct rxm_ep *rxm_ep = rx_buf->ep;
-	int ret;
 
 	memset(rx_buf, 0, sizeof(*rx_buf));
 	rx_buf->hdr = hdr;
 	rx_buf->hdr.state = RXM_RX;
 	rx_buf->ep = rxm_ep;
 
-	ret = fi_recv(rx_buf->hdr.msg_ep, &rx_buf->pkt, RXM_BUF_SIZE,
-		      rx_buf->hdr.desc, FI_ADDR_UNSPEC, rx_buf);
-	if (ret)
+	if (fi_recv(rx_buf->hdr.msg_ep, &rx_buf->pkt, RXM_BUF_SIZE,
+		    rx_buf->hdr.desc, FI_ADDR_UNSPEC, rx_buf)) {
 		FI_WARN(&rxm_prov, FI_LOG_EP_CTRL, "Unable to repost buf\n");
-	return ret;
+		return -FI_EAVAIL;
+	}
+	return FI_SUCCESS;
 }
 
 int rxm_ep_prepost_buf(struct rxm_ep *rxm_ep, struct fid_ep *msg_ep)
@@ -478,10 +478,10 @@ static int rxm_ep_peek_recv(struct rxm_ep *rxm_ep, fi_addr_t addr, uint64_t tag,
 			    0, NULL, rx_buf->pkt.hdr.data, rx_buf->pkt.hdr.tag);
 }
 
-static int rxm_ep_recv_common(struct rxm_ep *rxm_ep, const struct iovec *iov,
-			      void **desc, size_t count, fi_addr_t src_addr,
-			      uint64_t tag, uint64_t ignore, void *context,
-			      uint64_t flags, struct rxm_recv_queue *recv_queue)
+static ssize_t rxm_ep_recv_common(struct rxm_ep *rxm_ep, const struct iovec *iov,
+				  void **desc, size_t count, fi_addr_t src_addr,
+				  uint64_t tag, uint64_t ignore, void *context,
+				  uint64_t flags, struct rxm_recv_queue *recv_queue)
 {
 	struct rxm_recv_entry *recv_entry;
 	struct rxm_rx_buf *rx_buf;
@@ -518,7 +518,7 @@ static int rxm_ep_recv_common(struct rxm_ep *rxm_ep, const struct iovec *iov,
 	if (!recv_entry)
 		return -FI_EAGAIN;
 
-	recv_entry->count 	= count;
+	recv_entry->count 	= (uint8_t)count;
 	recv_entry->addr 	= src_addr;
 	recv_entry->context 	= context;
 	recv_entry->flags 	= flags;
@@ -663,7 +663,7 @@ static ssize_t rxm_rma_iov_init(struct rxm_ep *rxm_ep, void *buf,
 		rma_iov->iov[i].len = (uint64_t)iov->iov_len;
 		rma_iov->iov[i].key = fi_mr_key(mr[i]);
 	}
-	rma_iov->count = count;
+	rma_iov->count = (uint8_t)count;
 	return sizeof(*rma_iov) + sizeof(*rma_iov->iov) * count;
 }
 
@@ -671,7 +671,7 @@ static ssize_t rxm_rma_iov_init(struct rxm_ep *rxm_ep, void *buf,
 static ssize_t
 rxm_ep_send_common(struct fid_ep *ep_fid, const struct iovec *iov, void **desc,
 		   size_t count, fi_addr_t dest_addr, void *context,
-		   uint64_t data, uint64_t flags, uint64_t tag, int op,
+		   uint64_t data, uint64_t flags, uint64_t tag, uint8_t op,
 		   uint64_t comp_flags)
 {
 	struct util_cmap_handle *handle;
@@ -684,7 +684,7 @@ rxm_ep_send_common(struct fid_ep *ep_fid, const struct iovec *iov, void **desc,
 	size_t pkt_size = 0;
 	ssize_t size;
 	uint8_t progress = 0;
-	int ret;
+	ssize_t ret;
 
 	rxm_ep = container_of(ep_fid, struct rxm_ep, util_ep.ep_fid.fid);
 
@@ -699,11 +699,12 @@ rxm_ep_send_common(struct fid_ep *ep_fid, const struct iovec *iov, void **desc,
 		return -FI_EAGAIN;
 	}
 
-	if (!(tx_entry = rxm_tx_entry_get(&rxm_ep->send_queue)))
+	tx_entry = rxm_tx_entry_get(&rxm_ep->send_queue);
+	if (!tx_entry)
 		return -FI_EAGAIN;
 
 	tx_entry->ep = rxm_ep;
-	tx_entry->count = count;
+	tx_entry->count = (uint8_t)count;
 	tx_entry->context = context;
 	tx_entry->flags = flags;
 	tx_entry->tx_buf = tx_buf;
@@ -1120,7 +1121,8 @@ static int rxm_ep_bind(struct fid *ep_fid, struct fid *bfid, uint64_t flags)
 		if (ret)
 			return ret;
 
-		if (!(rxm_ep->util_ep.cmap = rxm_conn_cmap_alloc(rxm_ep)))
+		rxm_ep->util_ep.cmap = rxm_conn_cmap_alloc(rxm_ep);
+		if (!rxm_ep->util_ep.cmap)
 			return -FI_ENOMEM;
 		break;
 	case FI_CLASS_CQ:
@@ -1342,7 +1344,8 @@ int rxm_endpoint(struct fid_domain *domain, struct fi_info *info,
 	if (!rxm_ep)
 		return -FI_ENOMEM;
 
-	if (!(rxm_ep->rxm_info = fi_dupinfo(info))) {
+	rxm_ep->rxm_info = fi_dupinfo(info);
+	if (!rxm_ep->rxm_info) {
 		ret = -FI_ENOMEM;
 		goto err1;
 	}

--- a/prov/rxm/src/rxm_init.c
+++ b/prov/rxm/src/rxm_init.c
@@ -165,7 +165,8 @@ static int rxm_getinfo(uint32_t version, const char *node, const char *service,
 		for (cur = *info; cur; cur = cur->next) {
 			if (!OFI_CHECK_MR_LOCAL(cur))
 				continue;
-			if (!(dup = fi_dupinfo(cur))) {
+			dup = fi_dupinfo(cur);
+			if (!dup) {
 				fi_freeinfo(*info);
 				return -FI_ENOMEM;
 			}

--- a/prov/rxm/src/rxm_rma.c
+++ b/prov/rxm/src/rxm_rma.c
@@ -36,9 +36,9 @@
 typedef ssize_t rxm_rma_msg_fn(struct fid_ep *ep_fid,
 			       const struct fi_msg_rma *msg, uint64_t flags);
 
-static int rxm_ep_rma_common(struct fid_ep *msg_ep, struct rxm_ep *rxm_ep,
-			     const struct fi_msg_rma *msg, uint64_t flags,
-			     rxm_rma_msg_fn rma_msg, uint64_t comp_flags)
+static ssize_t rxm_ep_rma_common(struct fid_ep *msg_ep, struct rxm_ep *rxm_ep,
+				 const struct fi_msg_rma *msg, uint64_t flags,
+				 rxm_rma_msg_fn rma_msg, uint64_t comp_flags)
 {
 	struct rxm_domain *rxm_domain;
 	struct rxm_tx_entry *tx_entry;
@@ -46,7 +46,8 @@ static int rxm_ep_rma_common(struct fid_ep *msg_ep, struct rxm_ep *rxm_ep,
 	size_t i;
 	int ret;
 
-	if (!(tx_entry = rxm_tx_entry_get(&rxm_ep->send_queue)))
+	tx_entry = rxm_tx_entry_get(&rxm_ep->send_queue);
+	if (!tx_entry)
 		return -FI_EAGAIN;
 
 	memset(tx_entry, 0, sizeof(*tx_entry));
@@ -147,15 +148,15 @@ static ssize_t rxm_ep_read(struct fid_ep *ep_fid, void *buf, size_t len,
 	return rxm_ep_readmsg(ep_fid, &msg, rxm_ep_tx_flags(ep_fid));
 }
 
-static int rxm_ep_rma_inject(struct fid_ep *msg_ep, struct rxm_ep *rxm_ep,
-			     const struct fi_msg_rma *msg, uint64_t flags)
+static ssize_t rxm_ep_rma_inject(struct fid_ep *msg_ep, struct rxm_ep *rxm_ep,
+				 const struct fi_msg_rma *msg, uint64_t flags)
 {
 	struct rxm_tx_entry *tx_entry;
 	struct rxm_tx_buf *tx_buf;
 	struct fi_msg_rma msg_rma;
 	struct iovec iov;
 	size_t size;
-	int ret;
+	ssize_t ret;
 
 	size = ofi_total_iov_len(msg->msg_iov, msg->iov_count);
 
@@ -185,7 +186,8 @@ static int rxm_ep_rma_inject(struct fid_ep *msg_ep, struct rxm_ep *rxm_ep,
 		return -FI_EAGAIN;
 	}
 
-	if (!(tx_entry = rxm_tx_entry_get(&rxm_ep->send_queue))) {
+	tx_entry = rxm_tx_entry_get(&rxm_ep->send_queue);
+	if (!tx_entry) {
 		rxm_cq_progress(rxm_ep);
 		ret = -FI_EAGAIN;
 		goto err1;

--- a/src/common.c
+++ b/src/common.c
@@ -82,9 +82,9 @@ uint64_t fi_tag_format(uint64_t tag_bits)
 	return FI_TAG_GENERIC >> (ffsll(htonll(tag_bits)) - 1);
 }
 
-int fi_size_bits(uint64_t num)
+uint8_t fi_size_bits(uint64_t num)
 {
-	int size_bits = 0;
+	uint8_t size_bits = 0;
 	while (num >> ++size_bits);
 	return size_bits;
 }

